providers/yaml/src/main/java/org/jboss/resteasy/plugins/providers/YamlProvider.java

```diff
@@ -52,9 +73,19 @@ public Object readFrom(Class<Object> type, Type genericType, Annotation[] annota
       try {
          LogMessages.LOGGER.debugf("Provider : %s,  Method : readFrom", getClass().getName());
          if (isValidInternalType(type)) {

```

Comment:
This block, down to 87, apply only to lists, etc. So lists, etc., also have to be on the allowedList (unless isTypeCheckDisabled() is true). Is that what you want? 

Also, isTypeCheckDisabled() applies only for lists, etc. Again, is that what you want?

Reply:
They pass because they start with the package "java.". This does seem to be what we want because the issue only seems to appear in generic types.